src/pages/iou/steps/MoneyRequestAmountForm.js

```diff
@@ -0,0 +1,266 @@
+
+    const selectedAmountAsString = amount ? CurrencyUtils.convertToWholeUnit(currency, amount).toString() : '';
+
+    const [currentAmount, setCurrentAmount] = useState(selectedAmountAsString);
+    const [shouldUpdateSelection, setShouldUpdateSelection] = useState(true);
+
+    const [selection, setSelection] = useState({
+        start: selectedAmountAsString.length,
+        end: selectedAmountAsString.length,
+    });
+
+    /**
+     * Event occurs when a user presses a mouse button over an DOM element.
+     *
+     * @param {Event} event
+     * @param {Array<string>} nativeIds
+     */
+    const onMouseDown = (event, nativeIds) => {
+        const relatedTargetId = lodashGet(event, 'nativeEvent.target.id');
+        if (!_.contains(nativeIds, relatedTargetId)) {
+            return;
+        }
+        event.preventDefault();
+        if (!textInput.current) {
+            return;
+        }
+        if (!textInput.current.isFocused()) {
+            textInput.current.focus();
+        }
+    };
+
+    /**
+     * Convert amount to whole unit and update selection
+     *
+     * @param {String} currencyCode
+     * @param {Number} amountInCurrencyUnits
+     */
+    const saveAmountToState = (currencyCode, amountInCurrencyUnits) => {
+        if (!currencyCode || !amountInCurrencyUnits) {
+            return;
+        }
+        const amountAsStringForState = CurrencyUtils.convertToWholeUnit(currencyCode, amountInCurrencyUnits).toString();
+        setCurrentAmount(amountAsStringForState);
+        setSelection({
+            start: amountAsStringForState.length,
+            end: amountAsStringForState.length,
+        });
+    };
+
+    useEffect(() => {
+        saveAmountToState(currency, amount);
+        // we want to update the state only when the amount is changed
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [amount]);
+
+    /**
+     * Sets the state according to amount that is passed
+     * @param {String} newAmount - Changed amount from user input
+     */
+    const setNewAmount = (newAmount) => {
+        // Remove spaces from the newAmount value because Safari on iOS adds spaces when pasting a copied value
+        // More info: https://github.com/Expensify/App/issues/16974
+        const newAmountWithoutSpaces = MoneyRequestUtils.stripSpacesFromAmount(newAmount);
+        // Use a shallow copy of selection to trigger setSelection
+        // More info: https://github.com/Expensify/App/issues/16385
+        if (!MoneyRequestUtils.validateAmount(newAmountWithoutSpaces)) {
+            setCurrentAmount((prevAmount) => prevAmount);
+            setSelection((prevSelection) => ({...prevSelection}));
+            return;
+        }
+        setCurrentAmount((prevAmount) => {
+            setSelection((prevSelection) => getNewSelection(prevSelection, prevAmount.length, newAmountWithoutSpaces.length));
+            return MoneyRequestUtils.stripCommaFromAmount(newAmountWithoutSpaces);
```

Comment:
👋 Coming from https://github.com/Expensify/App/issues/23933
> Cursor jumps to next position when we use ',' in request money when language is English

This happens because we were updating the cursor position using the non-comma stripped amount and only after that strip the comma from the amount.
We resolved this by stripping comma from the amount first and updating the selection after that, i.e
```js
  setCurrentAmount((prevAmount) => {
      const strippedAmount = MoneyRequestUtils.stripCommaFromAmount(newAmountWithoutSpaces);
      setSelection((prevSelection) => getNewSelection(prevSelection, prevAmount.length, strippedAmount.length));
      return strippedAmount;
  });

Reply:
I think that makes sense but it's not a regression from this PR. That logic was copied from `src/pages/iou/steps/MoneyRequestAmount.js`,